library/Hasql/Pool.hs

```diff
@@ -17,16 +17,21 @@ import Hasql.Pool.Prelude
 import Hasql.Session (Session)
 import qualified Hasql.Session as Session
 
+data ReuseConnection = Keep | Close
+
 -- | A pool of connections to DB.
 data Pool = Pool
   { -- | Connection settings.
     poolFetchConnectionSettings :: IO Connection.Settings,
     -- | Avail connections.
     poolConnectionQueue :: TQueue Connection,
-    -- | Capacity.
+    -- | Remaining capacity.
+    -- The pool size limits the sum of poolCapacity, the length
+    -- of length poolConnectionQueue and the number of in-flight
+    -- connections.
     poolCapacity :: TVar Int,
-    -- | Alive.
-    poolAlive :: TVar Bool
+    -- | Whether to return a connection to the pool.
+    poolReuseToggle :: TVar (TVar ReuseConnection)
```

Comment:
I'm at a bit of a loss how to name this well. I went for very explicit naming for the purposes of making the PR understandable for now, hopefully we can find something better?